File: src/front-end-nextjs/hob-dashboard/lib/types/order.ts
export interface Order {
  orderId: string;
  customerId: string;
  customerName: string;
  orderDate: string;
  totalAmount: number;
  status: string;
  createdAt: string;
  updatedAt: string;
  sales: SaleSummary[];
}

export interface SaleSummary {
  saleId: string;
  productName: string;
  quantity: number;
  unitPrice: number;
  totalPrice: number;
}

export interface CreateOrderRequest {
  customerId: string;
  orderDate?: string;
  sales: CreateSaleItem[];
}

export interface CreateSaleItem {
  productName: string;
  quantity: number;
  unitPrice: number;
}

export interface UpdateOrderRequest {
  status: string;
}

File: src/HOB.Data/Entities/Order.cs
namespace HOB.Data.Entities;

public class Order
{
    public Guid OrderId { get; set; }
    public Guid CustomerId { get; set; }
    public DateTime OrderDate { get; set; }
    public decimal TotalAmount { get; set; }
    public string Status { get; set; } = "Pending";
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // Navigation properties
    public virtual Customer Customer { get; set; } = null!;
    public virtual ICollection<Sale> Sales { get; set; } = new List<Sale>();
}
